code/modules/recycling/disposal.dm

```diff
@@ -439,66 +439,62 @@
 	else
 		return ..(mover, target, height, air_group)
 
-/obj/machinery/disposal/MouseDrop_T(atom/dropping, mob/user)
-	if(istype(user, /mob/living/silicon/ai))
+/obj/machinery/disposal/MouseDrop_T(atom/movable/dropping, mob/user)
+
+	if(isAI(user))
+		return
+
+	//We are restrained, can't move or locked to something, this will compromise taking out the trash
+	if(user.restrained() || !user.canmove || user.locked_to)
 		return
 
-	if(!ismob(dropping))
-		if(istype(dropping, /obj/item))
-			if(!user.restrained() && user.canmove)
-				attackby(dropping, user)
+	if(istype(dropping, /obj/item))
 
+		if(dropping.locked_to) //Items can very specifically be locked to something, check that here
+			return
+
+		attackby(dropping, user)
 		return
 
 	var/locHolder = dropping.loc
 	var/mob/target = dropping
 
+	//Our target, now confirmed to be a mob, can't move or is locked to something, same thing
+	if(!target.canmove || target.locked_to)
+		return
+
 	if(target == user)
-		if(!user.restrained() && user.canmove && !user.locked_to)
-			target.visible_message("[target] starts climbing into the [src].", "You start climbing into the [src].")
-		else
-			return
+		target.visible_message("[target] starts climbing into \the [src].", "You start climbing into \the [src].")
+
 	else
-		if(isanimal(user))
-			return // animals cannot put mobs other than themselves into disposal
 
-		if(!user.restrained() && user.canmove)
-			if(target.locked_to)
-				return
```

Comment:
Before: If you couldn't move but weren't locked_to, someone else could disposal you.

Now: You can't be disposaled if you can't move.